aws-athena-connector/src/glue.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Transform Glue table schema into Data Studio Fields.
+ *
+ * For each column in Glue Table, it maps to one of the data types in Data Studio.
+ * Glue Table data types: https://docs.aws.amazon.com/athena/latest/ug/data-types.html
+ * NOTE: BINARY, ARRAY, MAP, STRUCT are ignored, since there's no corresponding data types.
+ *
+ * @param  {Object} table Table object from AWSGlue.GetTable.
+ * @return {Object} Data Studio Fields object.
+ */
+function glueTableToFields(table) {
+  var cc = DataStudioApp.createCommunityConnector();
+  var fields = cc.getFields();
+  var types = cc.FieldType;
+  var columns = table.PartitionKeys.concat(table.StorageDescriptor.Columns);
+
+  columns.forEach(function (column) {
+    // Set fields based on data type
+    var field;
+    switch (column.Type.toLowerCase()) {
+      case 'boolean':
+        field = fields.newDimension()
+          .setType(types.BOOLEAN);
+        break;
+      case 'tinyint':
+      case 'smallint':
+      case 'int':
+      case 'bigint':
+      case 'double':
+      case 'float':
+      case 'decimal':
+        field = fields.newMetric()
+          .setType(types.NUMBER);
+        break;
+      case 'char':
+      case 'varchar':
+      case 'string':
+      case 'timestamp':
+        field = fields.newDimension()
+          .setType(types.TEXT);
```

Comment:
Could this be `types.YEAR_MONTH_DAY`, or one of the other options instead of text?

Reply:
Yes I've updated to use `types.YEAR_MONTH_DAY_HOUR`